soc/xtensa/intel_adsp/common/include/soc.h

```diff
@@ -69,11 +69,11 @@ static ALWAYS_INLINE void z_idelay(int n)
 /* memcopy used by boot loader */
 static ALWAYS_INLINE void bmemcpy(void *dest, void *src, size_t bytes)
 {
-	uint32_t *d = (uint32_t *)dest;
-	uint32_t *s = (uint32_t *)src;
+	volatile uint32_t *d = (uint32_t *)dest;
+	volatile uint32_t *s = (uint32_t *)src;
 
 	z_xtensa_cache_inv(src, bytes);
-	for (size_t i = 0; i < (bytes >> 2); i++)
+	for (volatile size_t i = 0; i < (bytes >> 2); i++)
```

Comment:
You don't need the volatile attribute on the loop index, it's sufficient to use volatile on the pointers.

Reply:
It's not required; the code is just less optimal.